noir-projects/aztec-nr/aztec/src/state_vars/private_mutable.nr

```diff
@@ -9,22 +9,26 @@ use crate::oracle::notes::check_nullifier_exists;
 use crate::state_vars::storage::Storage;
 
 // docs:start:struct
-struct PrivateMutable<Note> {
-    context: Option<&mut PrivateContext>,
+struct PrivateMutable<Note, Context> {
+    context: Context,
     storage_slot: Field
 }
 // docs:end:struct
 
-impl<T> Storage<T> for PrivateMutable<T> {}
+impl<T, Context> Storage<T> for PrivateMutable<T, Context> {}
 
-impl<Note> PrivateMutable<Note> {
+impl<Note, Context> PrivateMutable<Note, Context> {
     // docs:start:new
     pub fn new(context: Context, storage_slot: Field) -> Self {
         assert(storage_slot != 0, "Storage slot 0 not allowed. Storage slots must start from 1.");
-        Self { context: context.private, storage_slot }
+        Self { context, storage_slot }
     }
     // docs:end:new
 
+    pub fn to_unconstrained(self) -> PrivateMutable<Note, ()> {
+        PrivateMutable { context: (), storage_slot: self.storage_slot }
+    }
+
```

Comment:
This might seem odd out of context, see my comment [here](https://github.com/AztecProtocol/aztec-packages/pull/6442/files#r1602298155).